src/dmd/cli.d

```diff
@@ -25,6 +25,7 @@ enum TargetOS
     macOS = 4,
     freeBSD = 8,
     solaris = 16,
+    dragonFlyBSD = 32,
```

Comment:
These questions/comments are not really related to this PR specifically:
What about the other OS's like NetBSD / OpenBSD etc ?
Would adding posix make sense ? 
Do we really need yet another spelling / enum for the target OS ?

Reply:
Well, the problem is that `dlang.org` needs to be able to set the target OS and the current `TARGET_` are constants. I'm not really happy with this approach.
@ZombineDev wanted to do a more general revamp and introduce target triples. In any case, I think currently there are only differences between `Windows` and `Posix`, so it might make sense to just use those two.

Reply:
@wilzbach : I did not see the original PR, or i would have made these comments there. 

The scattered nature and their repetition of these OS specific definitions is what throws me off a little to be honest. From a porting perspective it would be nice to see all OS specific stuff in one single file / directory, but that has been discussed before (over and over/ad nausea-um), so dropping that.

Except i just added one more 'exception' in line 215 (see above) so now there is three :-)